### PR TITLE
blockdev_mirror_cancel_running_job:Cancel a running mirror job

### PR DESCRIPTION
**Pull request by aliang123**
Cancel a running mirror job

Signed-off-by: Aihua Liang <aliang@redhat.com>
id:1985937

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_mirror_cancel_running_job.q35: PASS (42.29 s)

**Comment by aliang123** [comment]
@zhencliu @ngu-niny, please help to review, thanks.

File: qemu/tests/blockdev_mirror_cancel_running_job.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            self.main_vm, 'BLOCK_JOB_CANCELLED',
+            self.params.get_numeric('job_cancelled_timeout', 60),
+            device=self._jobs[0]
+        )
+        if event is None:
+            self.test.fail('Job failed to cancel')
+
+    def do_test(self):
+        self.blockdev_mirror()
+        self.cancel_job()
```
Comment (zhencliu) [inline qemu/tests/blockdev_mirror_cancel_running_job.py:22]: One concern, do we need to check when is the job running?

Comment (aliang123) [reply qemu/tests/blockdev_mirror_cancel_running_job.py:24]: OK, @zhenchao, have added the code to check if job in running status.

**Comment by zhencliu** [comment]
LGTM

**Comment by ngu-niny** [comment]
1-Host_RHEL.m8.u5.product_av.raw.virtio_blk.up.virtio_net.Guest.RHEL.9.0.0.ppc64le.io-github-autotest-qemu.blockdev_mirror_cancel_running_job Finshed 2021-08-03 04:08:19 2021-08-03 04:09:44 PASS 84.552527

Acked-by: Nini Gu <ngu@redhat.com>

**Pull request merged**